pkgs/tools/typesetting/tex/texlive/build-texlive-package.nix

```diff
@@ -202,5 +219,51 @@ let
       mkdir -p "$out"/share
       ln -s {"$texdoc"/doc,"$out"/share}/info
     '';
+
+  # most packages can be used as build inputs by adding their path to TEXMFAUXTREES
+  # exceptions are:
+  # - format binaries, which must be provided by a combined build input
+  # - hyphenation patterns, which cause formats to be rebuilt
+  # - occasional packages that look up in TEXMFROOT or similar
```

Comment:
Are there any fundamental problems with just adding format containers that are built with all the hyphenation patterns (and leave an escape hatch that drops the containers and runs fmtutil if someone wants to add their own custom hyphenation patterns)?

Reply:
arrgh, this wouldn't allow the paper size customization

Reply:
> format containers that are built with all the hyphenation patterns

That could be an option, just force all hyphenation patterns in every scheme. If the size is small I don't see why not, surely adding more patterns has no other side effect.

I would still build the formats by calling `buildEnv` recursively, that way we are sure that they use the same configuration, including paper size.